tests/test_data_io.py

```diff
@@ -70,6 +74,22 @@ def test_write_index(tmp_path_factory, dataframe, manifest):
         df = dd.read_parquet(fn / "index")
         assert len(df) == NUMBER_OF_TEST_ROWS
         assert list(df.columns) == ["id", "source"]
+        assert df.index.name == "uid"
+
+
+def test_rewrite_index(tmp_path_factory, dataframe, manifest):
+    """Test writing out the index subset without a dask index."""
```

Comment:
It took me a while to figure out what this test was doing. If I understand correctly, it checks that the index is created correctly when writing, right? Can you add some explanation on this in the docstring?

Same for the subset test below.